### DISCUSSION TITLE
Spieländerungen per MQTT

### DISCUSSION THREAD

**Discussion by msiggi** [OpenLigaDB/OpenLigaDB-Samples#77] [Show and tell]
Änderungen in den Spielergebnissen werden per MQTT auf den öffentlich verfügbaren Broker "broker.hivemq.com" gepushed. MQTT-Clients müssen sich zum Empfang der Daten diesem mit dem Topic "openligadb" subscriben und werden dann bei Ergebnisänderung mit dem match-Objekt benachrichtigt.

- Broker: broker.hivemq.com
- Port: 8883
- TLS-enabled
- Subscription-Topics:
  - openligadb/# - alle Änderungen aller Spiele
  - openligadb/bl1/# - alle Änderungen aller Spiele der 1. Fußball Bundesliga
  - openligadb/bl1/2024/# - alle Änderungen der Spiele der 1. Fußball Bundesliga der Saison 2024/2025
  - openligadb/bl1/2024/1 - alle Änderungen der Spiele des 1. Spieltages der 1. Fußball Bundesliga der Saison 2024/2025

Siehe auch #64 und #81

**Comment by xdream77** [comment]
Ich habe [dafür](https://jsr.io/@wgd/oldb@0.2.6) einen kleinen Javascript Adapter geschrieben. 
Vielleicht ist es ja hilfreich für den einen oder anderen. 
Funktioniert mit Node und Bun. Wahrscheinlich auch mit Deno, das habe ich aber nicht getestet. 

@msiggi wenn du das hier in die Beispiele integrieren möchtest, können wir das Repo auch gerne transferieren.

**Reply by msiggi** [reply]
Ja, gerne. Vorzugsweise erstellst Du einen Fork dieses Repositories, integrierst Deinen Code und schickst mir einen Pull-Request, wäre das Ok für Dich?

**Reply by xdream77** [reply]
Das klingt nach einem Plan. Lass mich noch ein bissel testen vorher, damit ich nicht direkt 700 Merge-requests stelle :-) 

Heute bekomme ich z.B. mitunter Fehler von HiveMQ.

`ErrorWithReasonCode: Connection refused: Not authorized code: "5"`
Und `Not authorized` dürfte da gar nicht kommen.

**Reply by msiggi** [reply]
Ja, temporär habe ich das auch schon gesehen. Gibt es evtl. einen zuverlässigeren gratis Broker?

**Reply by xdream77** [reply]
Das weiß ich leider nicht. Es ist auch das erste mal, dass ich mit hiveMQ arbeite.

**Comment by hokascha** [comment]
Bist du auf shared hosting oder warum nicht einen eigenen Broker auf dem gleichen Server? Mosquito braucht nicht viel Ressourcen.

**Reply by msiggi** [reply]
Es könnte problematisch sein, einen öffentlich bekannten MQTT-Broker ohne notwendige Accountdaten zu betreiben. Ich könnte mir vorstellen, dass dieser schnell für Daten aller Art benutzt werden würde. Und die Implementierung einer Nutzerverwaltung bzw. die Integration der OpenLigaDB-Userverwaltung für die MQTT-Anmeldung ist mir aktuell zu aufwendig.

**Comment by hokascha** [comment]
Ok, nachvollziehbar. Also Aufruf an alle: freien, zuverlässigen Broker finden oder darüber nachdenken, wie man einen eigenen, halbwegs sicheren Broker betreiben kann.
